### ISSUE TITLE
[Bug🐛] Fix test module ordering in the admin topic adapter

### ISSUE THREAD

**Issue by mxsm** [mxsm/rocketmq-rust#10458]
### Prerequisites

- [x] I have searched existing issues and discussions
- [x] This is a bug, not a question (use Discussions for questions)

### Environment

- Package: `rocketmq-admin-core`, in the root Cargo workspace.
- Source inspected: main at `8572512a49b9e854c50751e7930b87a47b8a86aa`.
- Reported toolchain: Clippy 1.95.0 with `-D warnings`.
- CI runner OS and run URL were not supplied.

### Bug Description

In `rocketmq-tools/rocketmq-admin/rocketmq-admin-core/src/client_adapter/topic.rs`, the `#[cfg(test)] mod tests` starts at line 696, but the production `impl crate::core::topic::TopicSkipMutationAdmin for AdminSession` follows it at line 757. This triggers `clippy::items_after_test_module` when the library test target is checked.

[Source: test module and following implementation](https://github.com/mxsm/rocketmq-rust/blob/8572512a49b9e854c50751e7930b87a47b8a86aa/rocketmq-tools/rocketmq-admin/rocketmq-admin-core/src/client_adapter/topic.rs#L695).

### Steps to Reproduce

From the repository root, using the toolchain selected by `rust-toolchain.toml`:

```bash
cargo clippy -p rocketmq-admin-core --features client-adapter --lib --tests --no-deps -- -D warnings
```

Enable `client-adapter`: this feature gates the affected topic adapter module, so a default-feature check will not cover it.

### Expected vs Actual

**Expected:** the affected library test target passes the item-ordering lint.

**Actual:** the supplied CI report shows a compilation failure with exit code 101:

### Logs/Screenshots

```text
error: items after a test module
   --> rocketmq-tools/rocketmq-admin/rocketmq-admin-core/src/client_adapter/topic.rs:696:1
696 | mod tests {
...
757 | impl crate::core::topic::TopicSkipMutationAdmin for AdminSession {
    = note: -D clippy::items-after-test-module implied by -D warnings
error: could not compile rocketmq-admin-core (lib test) due to 1 previous error
```

This diagnostic was supplied by the maintainer. The module order and feature gate were confirmed in source; Clippy was not rerun during issue preparation.

### Additional Context

This is a small, mechanical task for a first-time contributor. Please comment before starting so contributors can coordinate.

#### Requested change

Move the entire `TopicSkipMutationAdmin for AdminSession` implementation immediately before `#[cfg(test)] mod tests`, leaving the test module at the end of the file.

Only change `rocketmq-tools/rocketmq-admin/rocketmq-admin-core/src/client_adapter/topic.rs`. Preserve the implementation body, all three existing tests, and feature gates. Do not add a lint allowance, alter CI settings, or split the file into new modules.

#### Acceptance criteria

- [ ] The test module is the final item in the file.
- [ ] The trait implementation and existing tests retain their behavior.
- [ ] The scoped Clippy command above no longer reports this lint.
- [ ] Package formatting and the existing focused tests pass.

Run these additional checks from the repository root:

```bash
cargo fmt -p rocketmq-admin-core -- --check
cargo test -p rocketmq-admin-core --features client-adapter --lib client_adapter::topic::tests
git diff --check
```

These are validation commands for the future contribution, not results already obtained. No new regression test is needed for this code-order-only change.

Duplicate checks found closed issue #9989 for the same lint in `rocketmq-store/src/factory.rs`; that is a different file and fix. No matching open issue or PR was found for this topic adapter.
